### ISSUE TITLE
`typing_extensions.TypeVar` instance becomes `typing.TypeVar`

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pyright#6241]
**Describe the bug**
When typing a callable to accept a `typing_extensions.TypeVar` instance as an argument, pyright seemed to transform either the annotation or the argument to `typing.TypeVar`, causing a type checker error. In the example below, it seems that `T` is being transformed into a `typing.TypeVar` as Pyright seems to say it has no `__default__` attribute defined.

**Code or Screenshots**

```python
from typing_extensions import TypeVar


def get_typevar_default(tv: TypeVar):
    return tv.__default__


T = TypeVar("T")

get_typevar_default(T) # fails
T.__default__ # fails
```

```py
test.py
  test.py:10:21 - error: Argument of type "TypeVar" cannot be assigned to parameter "tv" of type "TypeVar" in function "get_typevar_default"
    "typing.TypeVar" is incompatible with "typing_extensions.TypeVar" (reportGeneralTypeIssues)
  test.py:11:3 - error: Cannot access member "__default__" for type "TypeVar"
    Member "__default__" is unknown (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
pyright 1.1.333

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

FWIW, `TypeVar.__default__` is proposed in the draft spec [PEP 696](https://peps.python.org/pep-0696/). This PEP has not yet been accepted by the Python steering council, so it could change or be rejected. Pyright and `typing_extensions` contain provisional support for the functionality, but I recommend not using it in production code until the PEP is ratified. If the PEP is rejected, it will be removed from Pyright and the `typing_extensions` library without notice.

**Comment by NCPlayz** [comment]
Thanks for letting me know, I just thought to use it as an example. I actually encountered the error while trying to solve another typing-related issue :grin:

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.334, which I just published. It will also be included in a future release of pylance.
